### PR TITLE
add: 补充MediaSource C API

### PR DESCRIPTION
**Pull request by lidaofu-hub**
补充MediaSource C API  获取源地址 获取源类型  获取创建时间戳

### CODE REVIEW

File: api/source/mk_events_objects.cpp [hunk 1]
```diff
@@ -182,7 +182,7 @@ API_EXPORT const char* API_CALL mk_media_source_get_vhost(const mk_media_source
 }
 API_EXPORT const char* API_CALL mk_media_source_get_app(const mk_media_source ctx){
     assert(ctx);
-    MediaSource *src = (MediaSource *)ctx;
+    MediaSource  *src = (MediaSource *)ctx;
```
Comment (suzp1984) [inline api/source/mk_events_objects.cpp:185]: extra useless white space

Comment (lidaofu-hub) [reply api/source/mk_events_objects.cpp]: Thank you for the reminder. It has been removed

File: api/source/mk_events_objects.cpp [hunk 2]
```diff
@@ -228,6 +228,30 @@ API_EXPORT int API_CALL mk_media_source_broadcast_msg(const mk_media_source ctx,
     return src->broadcastMessage(any);
 }
 
+API_EXPORT const char* API_CALL mk_media_source_get_origin_url(const mk_media_source ctx) {
+    assert(ctx);
+    MediaSource *src = (MediaSource *)ctx;
+    return _strdup(src->getOriginUrl().c_str());
```
Comment (suzp1984) [inline api/source/mk_events_objects.cpp:234]: `#include <string.h>`?

Comment (lidaofu-hub) [reply api/source/mk_events_objects.cpp:238]: Thank you, I have found the reason why the non Windows environment did not compile successfully

Comment (suzp1984) [reply api/source/mk_events_objects.cpp:238]: Are you talking `_WIN32` inside `api/source/mk_util.cpp`? 

https://stackoverflow.com/questions/7582394/strdup-or-strdup
It seems `_strdup` is WIN standards, then it's duplicated defined here and `api/source/mk_util.cpp`, maybe extract this Macro to a utility header? (`api/include/mk_common.h`?)

Comment (lidaofu-hub) [reply api/source/mk_events_objects.cpp:234]: Thank you, there are indeed duplicate definitions. I have defined them in `api/include/mk_common.h`

**Pull request merged**